### PR TITLE
Retry when WAL segment has not been found

### PR DESCRIPTION
**Pull request by iskakaushik**
[Debezium](https://github.com/debezium/debezium/blob/db0cfea550308e7386413db25115d531a4c0a61c/debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java#L584) also does something similar. Awaiting confirmation from Postgres experts on this but so far what we have seen from the field indicates that this error is OK.

This PR also does some minor refactors and cleaning up of error messages that looked very similar.

### CODE REVIEW

File: flow/connectors/postgres/cdc.go [hunk 1]
```diff
@@ -324,6 +353,13 @@ func (p *PostgresCDCSource) consumeStream(
 		}
 
 		if errMsg, ok := rawMsg.(*pgproto3.ErrorResponse); ok {
+			if errMsg.Severity == "ERROR" && errMsg.Code == "XX000" {
```
Comment (serprex) [inline flow/connectors/postgres/cdc.go:356]: No retry limit?

Comment (iskakaushik) [reply flow/connectors/postgres/cdc.go:359]: added retries

**Review by serprex — APPROVED** [review]

**Comment by heavycrystal** [comment]
can we also log the segment that is erroring?

**Comment by iskakaushik** [comment]
> can we also log the segment that is erroring?

it's getting logged higher up in the stack

**Pull request merged**